### ISSUE TITLE
Resume training after certain iterations

### ISSUE THREAD

**Issue by Atul997** [hpanwar08/detectron2#37]
I am using Colab for training my model and as the time limit gets over, training stops. I have checkpoints saved after 500 iterations. I want to resume my training given a checkpoint in the output folder but every time it starts training from iteration 0. I have opened the `train_net_dla.py` file and training the model, below is configurations and steps for training

    cfg = get_cfg()
    # mask rcnn resnet101
    # mask rcnn resnext
    cfg.merge_from_file("./configs/DLA_mask_rcnn_X_101_32x8d_FPN_3x.yaml")
    cfg.MODEL.WEIGHTS = "/content/drive/MyDrive/layout/model_final_trimmed.pth"
    output_dir = "./output"
    cfg.OUTPUT_DIR = output_dir
    #os.makedirs(output_dir, exist_ok=True)
    cfg.TEST.EVAL_PERIOD = 200
    cfg.SOLVER.CHECKPOINT_PERIOD= 500
    #logger.info(cfg)

    # serialize the training config
    cfg_str = cfg.dump()
    with open(os.path.join(cfg.OUTPUT_DIR, "train_config.yaml"), "w") as f:
        f.write(cfg_str)
        f.close()
    model = build_model(cfg)
    checkpointer = DetectionCheckpointer(model).load(cfg.MODEL.WEIGHTS)
    checkpointer = DetectionCheckpointer(model, save_dir=output_dir)
    checkpointer = PeriodicCheckpointer(checkpointer, period=200,max_iter = 500)
     checkpointer.save("model")
    trainer = COCOTrainer(cfg)
    trainer.resume_or_load(resume=True)
    trainer.train()

**Comment by hpanwar08** [comment]
You could use `DefaultTrainer` without the need for `DetectionCheckpointer` and `PeriodicCheckpointer`. That should load the previously saved model

**Comment by Atul997** [comment]
Okay. Thanks :)
